components/script/dom/htmlmediaelement.rs

```diff
@@ -296,6 +296,41 @@ impl From<MediaStreamOrBlob> for SrcObject {
     }
 }
 
+struct DroppableField {
+    /// Player Id reported the player thread
+    id: Cell<u64>,
+    glplayer_chan: Option<GLPlayerPipeline>,
+    /// Media controls id.
+    /// In order to workaround the lack of privileged JS context, we secure the
+    /// the access to the "privileged" document.servoGetMediaControls(id) API by
+    /// keeping a whitelist of media controls identifiers.
+    media_controls_id: DomRefCell<Option<String>>,
+    root: Option<DomRoot<Document>>,
```

Comment:
Is there no problem that struct takes `DomRoot`?

Reply:
@jdm 
Currently, `DroppableField` has `DomRoot<Document>` struct but I think this implementation is incorrect.   So I think `DomRoot<Document>` needs to be removed from `DroppableField`.
Is there how to execute `root.unregister_media_controls()` in Drop trait when `DroppableField` does not have `DomRoot<Document>` ?

Reply:
We can store a PipelineId instead, and add a static method to ScriptThread (like [this](https://github.com/servo/servo/blob/d35f1ee5e74c1b7d0efffff13717ef70f738f5ad/components/script/script_thread.rs#L852-L854)) that calls ScriptThread::find_document and unregisters the media controls if it's found.

Reply:
I fixed this review. https://github.com/servo/servo/pull/28302/commits/1b2c726b898b280d4486cdbe4b801273fc6e4018